### PR TITLE
Using Client Certificate for 1.2 tests as IAM wasn't introduced till 1.3.

### PR DESCRIPTION
**Pull request by maisem**
@cjcullen 

cc: @fabioy @spxtr @ixdy @rmmh

---
This change is [Reviewable](https://reviewable.io/reviews/kubernetes/test-infra/548)

### CODE REVIEW

File: jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e-gke.yaml [hunk 1]
```diff
@@ -626,15 +622,23 @@
             version-cluster: '1.2'
             version-client: '1.3'
             version-infix: '1-2-1-3'
+            version-env: |
+                # 1.2 doesn't support IAM, so we should use cert auth.
```
Comment (cjcullen) [inline jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e-gke.yaml]: This one shouldn't be necessary. 1.3 client should be able to handle everything fine.

**Comment by cjcullen** [comment]
LGTM other than the one that should be unnecessary.

**Comment by ixdy** [comment]
Is this the cause of the "Please enter Username: Please enter Password: Error from server: the server has asked for the client to provide credentials" error messages we're seeing?

**Comment by maisem** [comment]
@ixdy  - yeah.

**Comment by maisem** [comment]
Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion.

---

*[jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e-gke.yaml, line 626 at r1](https://reviewable.io:443/reviews/kubernetes/test-infra/548#-KRQqvo8aXkU0b6hyKqi-r1-626:-KRZZb-k6Yqbjuv5MxZA:b-896fix) ([raw file](https://github.com/kubernetes/test-infra/blob/e549836206d0d46b9f8012691f14586f198fbf43/jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e-gke.yaml#L626)):*
<details><summary><i>Previously, cjcullen (CJ Cullen) wrote…</i></summary>
> This one shouldn't be necessary. 1.3 client should be able to handle everything fine.
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/test-infra/548)*

**Comment by cjcullen** [comment]
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/test-infra/548)*

**Pull request merged**